File: classes/201-f25/Comparing-Python-and-Kotlin/lists.py
"""
   lists.py
   Jeff Ondich, 2013-01-05
   Anna Rafferty

   Modified by Tanya Amert for Spring 2025

   Demonstrates a few operations with lists.
   
   Intended as the Python half of parallel examples in Python and
   Kotlin. See Lists.kt.
"""

# Initialize
animal_list = ['goat', 'moose', 'emu', 'kudu']

# Add an item
animal_list.append('coatimundi')

# Insert an item in the third position
animal_list.insert(2, 'gnat')

# Iterate over the items in the list using a for loop
print('Animals iterated using a for/in loop')
for animal in animal_list:
    print(animal)

print()
print('Animals iterated using an index')
for k in range(len(animal_list)):
    print(animal_list[k])

File: classes/201-f25/Comparing-Python-and-Kotlin/Lists.kt
/*
 * Lists.kt
 * Tanya Amert, 2025-03-31
 * Based on Lists.java by Jeff Ondich, Anna Rafferty
 *
 * Demonstrates a few operations with Kotlin lists.
 *
 * This is the Kotlin half of a pair of parallel examples in Python and Kotlin.
 * See lists.py.
 */

fun main() {
    // Initialize
    val animalList: MutableList<String> = mutableListOf("goat", "moose", "emu", "kudu")

    // Add an item
    animalList.add("beaver")

    // Insert an item in the third position
    animalList.add(2, "gnat")

    // Iterate over the items in the list using a for loop
    println("Animals iterated using a for/in loop")
    for (animal: String in animalList) {
        println(animal)
    }

    println()
    println("Animals iterated using an index")
    for (k: Int in 0..<animalList.count()) {
        println(animalList[k])
    }
}